hack/ensure-golangci-lint.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+set -euo pipefail
+
+VERSION=1.41.1
+GOOS=$(go env GOOS)
+TARNAME=golangci-lint-$VERSION-$GOOS-amd64.tar.gz
+URL=https://github.com/golangci/golangci-lint/releases/download/v$VERSION/$TARNAME
+GOLANGCI_LINT=bin/golangci-lint
+
+case $GOOS in
+    linux)
+        CHECKSUM=2ad1b2313f89d8ecb35f2e4d2463cf5bc1e057518e4c523565f09416046c21f7
+        ;;
+    darwin)
+        CHECKSUM=458f15c43f72b0bd905ff165c42cf474e5510f91f4a10ae025e42e22fe8b578f
```

Comment:
anybody has an m1 on our team? I think for mac we need to support ARM+AMD64 somehow

Reply:
I'm not sure but this will have to be done at some point. For now just `amd64` support is good enough.